src/calendar.cpp

```diff
@@ -59,9 +59,10 @@ double default_daylight_level()
 
 moon_phase get_moon_phase( const time_point &p )
 {
-    //One full phase every 1 rl months = 1/3 season length
-    const time_duration moon_phase_duration = calendar::season_length() / 3.0;
-    //Switch moon phase at noon so it stays the same all night
+    static constexpr time_duration synodic_month = 29.530588853 * 1_days;
+    const time_duration moon_phase_duration =
+        calendar::season_from_default_ratio() * synodic_month;
+    // Switch moon phase at noon so it stays the same all night
```

Comment:
This comment will not be necessarily true with non synchronous moon cycles.

Reply:
You're right that this was broken, but I think it was already broken before my changes.  In any case, I've now fixed it, and added a test.